src/useBookReviewApp.ts

```diff
@@ -0,0 +1,160 @@
+import { ChangeEvent, useEffect, useState } from "react";
+import axios, { AxiosError } from "axios";
+import { makeStyles } from "@material-ui/core";
+import { fetchReviews, postReview, putReview } from "./api";
+import { BasicItemProps } from "./components/Item";
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    flexGrow: 1,
+  },
+  paper: {
+    height: 140,
+    width: 100,
+  },
+  control: {
+    padding: theme.spacing(2),
+  },
+}));
+
+export function useBookReviewApp() {
+  const [items, setItems] = useState<BasicItemProps[]>([]);
+  const [isModalOpened, setIsModalOpened] = useState(false);
+  const [id, setId] = useState("0");
+  const [formData, setFormData] = useState({
+    title: "",
+    score: 5,
+    body: "",
+    reviewer: "",
+  });
+  const [isEdit, setIsEdit] = useState(false);
+  const [errors, setErrors] = useState<string[]>([]);
+
+  function openModal() {
+    setIsModalOpened(true);
+  }
+
+  function openEditModal(item: BasicItemProps) {
+    setId(item.id);
+    setFormData({
+      title: item.title,
+      score: item.score,
+      body: item.body,
+      reviewer: item.reviewer,
+    });
+    setIsEdit(true);
+    openModal();
+  }
+
+  function closeModal() {
+    setIsModalOpened(false);
+    setId("0");
+    setFormData({
+      title: "",
+      score: 5,
+      body: "",
+      reviewer: "",
+    });
+    setIsEdit(false);
+    setErrors([]);
+  }
+
+  function changeTitle(e: ChangeEvent<HTMLInputElement>) {
+    setFormData({ ...formData, title: e.target.value });
+  }
+
+  function changeScore(e: ChangeEvent<HTMLInputElement>) {
+    setFormData({ ...formData, score: Number(e.target.value) });
+  }
+
+  function changeBody(e: ChangeEvent<HTMLTextAreaElement>) {
+    setFormData({ ...formData, body: e.target.value });
+  }
+
+  function changeReviewer(e: ChangeEvent<HTMLInputElement>) {
+    setFormData({ ...formData, reviewer: e.target.value });
+  }
```

Comment:
```suggestion
  function handleReview(e: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>): void {
    setFormData({ ...formData, [e.target.name]: e.target.value });
  end
```

のように一つの関数にまとめて、各 input 側で `name="xxx"` をセットする、としてもいいです